### PR TITLE
Fire a deep link event when deep link action is called

### PR DESCRIPTION
**Pull request by gatlinhebert**
Enables the ability to handle deep links triggered from in-app messages, landing pages, and the message center.

### CODE REVIEW

**Comment by rlepinski** [comment]
Does the event get delivered when starting the app from a notification that contains a deep link?

**Comment by gatlinhebert** [comment]
It fails when launching from a notification as the listener callback is unavailable.

**Comment by rlepinski** [comment]
We probably need to add a map of pending events if the listener is not there, then fire all those events in the map as soon as the listener is available.

File: Example/index.html [hunk 1]
```diff
@@ -65,6 +73,7 @@
           // Reregister for urbanairship events if they were removed in pause event
           document.addEventListener("urbanairship.registration", onRegistration, false)
           document.addEventListener("urbanairship.push", onPushReceived, false)
+          document.addEventListener("urbanairship.deeplink", handleDeepLink, false)
```
Comment (marc-scig) [inline Example/index.html]: the one above has an underscore, is that intentional?

Comment (rlepinski) [inline Example/index.html]: nope, good catch

File: Example/index.html [hunk 2]
```diff
@@ -74,6 +83,7 @@
           // Remove urbanairship events.  Important on android to not receive push in the background.
           document.removeEventListener("urbanairship.registration", onRegistration, false)
           document.removeEventListener("urbanairship.push", onPushReceived, false)
+          document.removeEventListener("urbanairship.deeplink", handleDeepLink, false)
```
Comment (marc-scig) [inline Example/index.html]: samesies

**Comment by marc-scig** [comment]
🍻  🐻  🐗  📊  🍔

File: Example/index.html [hunk 3]
```diff
@@ -45,9 +45,17 @@
           }
         }
 
+        //Deep link callback
+        var handleDeepLink = function(event)  {
+          if (event.value) {
```
Comment (gatlinhebert) [inline Example/index.html]: if (event.deepLink)

**Comment by gatlinhebert** [comment]
+1 🎉

**Pull request merged**